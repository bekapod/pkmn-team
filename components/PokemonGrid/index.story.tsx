// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import PokemonGrid from ".";
import PokemonCard from "../PokemonCard";

const pokemon = (
  <PokemonCard
    pokemon={{
      id: "25",
      name: "pikachu",
      pokedexId: 25,
      sprite: "25.png",
      types: [{ name: "ELECTRIC" }]
    }}
  />
);

storiesOf("PokemonGrid", module)
  .add("with pokemon", () => (
    <PokemonGrid>
      {pokemon}
      {pokemon}
      {pokemon}
      {pokemon}
      {pokemon}
    </PokemonGrid>
  ))
  .add("empty", () => <PokemonGrid />);
