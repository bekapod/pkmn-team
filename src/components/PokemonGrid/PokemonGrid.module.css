.container {
  display: grid;
  grid-gap: var(--spacing-5);
  grid-template-columns: repeat(1, 1fr);
  width: 100%;

  @media (--medium-viewport) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media (--large-viewport) {
    grid-template-columns: repeat(3, 1fr);
  }
}
