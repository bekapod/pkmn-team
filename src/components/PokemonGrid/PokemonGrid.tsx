import type { FunctionComponent, ComponentPropsWithRef } from 'react';
import cx from 'classnames';

export const PokemonGrid: FunctionComponent<ComponentPropsWithRef<'div'>> = ({
  className,
  ...props
}) => (
  <div
    className={cx(
      'grid',
      'gap-5',
      'grid-cols-1',
      'w-full',
      'md:auto-cols-min',
      className
    )}
    {...props}
  />
);
