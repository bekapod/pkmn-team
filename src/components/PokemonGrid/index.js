// @flow
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";
import { media } from "../../helpers/media";

const PokemonGrid = styled.div`
  display: grid;
  grid-gap: ${variables.gutters.grid}px;
  grid-template-columns: repeat(1, 1fr);
  width: 100%;

  ${media.medium`
    grid-template-columns: repeat(2, 1fr);
  `};

  ${media.large`
    grid-template-columns: repeat(3, 1fr);
  `};
`;

export default PokemonGrid;
