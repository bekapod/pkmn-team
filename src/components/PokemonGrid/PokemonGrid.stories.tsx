import { Meta } from '@storybook/react/types-6-0';
import { PokemonCard } from '../PokemonCard';
import { PokemonGrid } from './PokemonGrid';

export default {
  title: 'Components/PokemonGrid',
  component: PokemonGrid
} as Meta;

export const Standard = (): JSX.Element => (
  <PokemonGrid>
    <PokemonCard
      pokemon={{
        _id: '25',
        _ts: '',
        learnableMoves: [],
        name: 'Pikachu',
        slug: 'pikachu',
        pokedexId: 25,
        sprite: '25.png',
        types: [
          {
            name: 'Electric',
            slug: 'electric',
            _id: '',
            _ts: '',
            moves: [],
            pokemon: []
          }
        ]
      }}
    />
    <PokemonCard
      pokemon={{
        _id: '25',
        _ts: '',
        learnableMoves: [],
        name: 'Pikachu',
        slug: 'pikachu',
        pokedexId: 25,
        sprite: '25.png',
        types: [
          {
            name: 'Electric',
            slug: 'electric',
            _id: '',
            _ts: '',
            moves: [],
            pokemon: []
          }
        ]
      }}
    />
    <PokemonCard
      pokemon={{
        _id: '25',
        _ts: '',
        learnableMoves: [],
        name: 'Pikachu',
        slug: 'pikachu',
        pokedexId: 25,
        sprite: '25.png',
        types: [
          {
            name: 'Electric',
            slug: 'electric',
            _id: '',
            _ts: '',
            moves: [],
            pokemon: []
          }
        ]
      }}
    />
    <PokemonCard
      pokemon={{
        _id: '25',
        _ts: '',
        learnableMoves: [],
        name: 'Pikachu',
        slug: 'pikachu',
        pokedexId: 25,
        sprite: '25.png',
        types: [
          {
            name: 'Electric',
            slug: 'electric',
            _id: '',
            _ts: '',
            moves: [],
            pokemon: []
          }
        ]
      }}
    />
    <PokemonCard
      pokemon={{
        _id: '25',
        _ts: '',
        learnableMoves: [],
        name: 'Pikachu',
        slug: 'pikachu',
        pokedexId: 25,
        sprite: '25.png',
        types: [
          {
            name: 'Electric',
            slug: 'electric',
            _id: '',
            _ts: '',
            moves: [],
            pokemon: []
          }
        ]
      }}
    />
  </PokemonGrid>
);
