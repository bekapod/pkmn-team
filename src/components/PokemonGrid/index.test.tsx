import React from "react";
import renderer from "react-test-renderer";
import PokemonGrid from ".";

describe("<PokemonGrid />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(
      <PokemonGrid>
        <div>Pokemon 1</div>
        <div>Pokemon 2</div>
      </PokemonGrid>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
