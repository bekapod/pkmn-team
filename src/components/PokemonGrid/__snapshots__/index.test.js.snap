// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonGrid /> renders without crashing 1`] = `
<div
  className="PokemonGrid-griwyh-0 lnIMTc"
>
  <div>
    Pokemon 1
  </div>
  <div>
    Pokemon 2
  </div>
</div>
`;
